### PR TITLE
Add aggregate function tests. Use CVL for expected output.

### PR DESCRIPTION
**Pull request by antvaset**
This PR adds new conformance tests for aggregate functions. I picked all the ones from https://github.com/cqframework/clinical_quality_language/blob/7e91478/Src/java/engine/src/test/resources/org/opencds/cqf/cql/engine/execution/CqlAggregateFunctionsTest.cql which didn't exist yet here.

Using CVL (literals only) for the expected output values, building on top of https://github.com/cqframework/cql-tests/pull/29. Focusing only on the `tests/cql/CqlAggregateFunctionsTest.xml` tests.

### CODE REVIEW

**Review by brynrhodes — APPROVED** [review]

File: tests/cql/CqlAggregateFunctionsTest.xml [hunk 1]
```diff
@@ -68,13 +72,23 @@
...
+		</test>
 	</group>
 	<group name="Avg">
 		<test name="AvgTest1">
 			<expression>Avg({ 1.0, 2.0, 3.0, 6.0 })</expression>
 			<output>3.0</output>
 		</test>
 	</group>
+	<group name="Product">
+		<test name="Product_Long">
```
Comment (suyashkumar) [inline tests/cql/CqlAggregateFunctionsTest.xml:87]: nit: I don't think underscores are used anywhere else for test names in this repo, so consider remaining consistent 

```suggestion
		<test name="ProductLong">
```

I personally don't feel that strongly though.

File: tests/cql/CqlAggregateFunctionsTest.xml [hunk 2]
```diff
@@ -68,13 +72,23 @@
...
+			<expression>Product({5L, 4L, 5L})</expression>
+			<output>100L</output>
+		</test>
```
Comment (suyashkumar) [inline tests/cql/CqlAggregateFunctionsTest.xml:90]: should we add other product tests (e.g. for other overloads, null input, etc) or should that come in another PR? (same comment for the other operators)

**Review by suyashkumar — APPROVED** [review]
LGTM, thanks! Just had a couple random comments.

File: tests/cql/CqlAggregateFunctionsTest.xml (hunk 2 above)
Comment (evan-gordon) [reply tests/cql/CqlAggregateFunctionsTest.xml:90]: +1 here if we're going to add this test group we should have a few more edge cases covered

**Review by raleigh-g-thompson — CHANGES_REQUESTED** [review]
Main branch has changed and there is now a merge conflict.  Resolve merge conflict before approval.
